internal/commands/gatekeeper.go

```diff
@@ -0,0 +1,64 @@
+package commands
+
+import (
+	"path/filepath"
+	"strings"
+)
+
+// Matchers are all of the matchers that can be applied to constraints.
+type Matchers struct {
+	KindMatchers []KindMatcher
+}
+
+// KindMatcher are the matchers that are applied to constraints.
+type KindMatcher struct {
+	APIGroup string
+	Kind     string
+}
+
+// GetMatchersFromComments returns all of the matchers found in the collection of comments.
+func GetMatchersFromComments(comments []string) Matchers {
+	var matchers Matchers
+	for _, comment := range comments {
+		if strings.Contains(strings.ToLower(comment), "@kinds") {
+			matchers.KindMatchers = getKindMatchers(comment)
+		}
+	}
+
+	return matchers
+}
+
+// GetNameFromPath returns the name of the resource based on its file path.
+func GetNameFromPath(path string) string {
+	name := filepath.Base(filepath.Dir(path))
+	name = strings.ReplaceAll(name, "-", " ")
+	name = strings.Title(name)
+
+	return name
+}
+
+// GetKindFromPath returns the kind of the resource based on its file path.
+func GetKindFromPath(path string) string {
+	name := GetNameFromPath(path)
+	kind := strings.ReplaceAll(name, " ", "")
+
+	return kind
+}
```

Comment:
Why are we making all of these public when they aren't used outside the package?

Reply:
`getNameFromPath` definely should not have been, thats only used within `gatekeeper.go` -- the others are set to public as I see those as "candidates" for being public when the package becomes a reality. Since we're inside of `internal` it doesn't really expose much, and no one should be consuming `commands`.

But the root is just playing with the idea of "I set this to public, but should it be?". If everything is private, it doesn't feel like it should've even been put into its own file.

Reply:
The `commands` package shouldn't really export anything, that just feels wrong to me. The package should deal almost exclusively with handling input from the command line, calling functions from other packages to perform the work, and then writing the output. If they are worth separating into a new file and making them public, they're probably worth separating into a new package.

Reply:
They aren't _truly_ public because everything is still underneath the parent level `internal` folder. I agree with you that the commands package should not have public methods and other files should not import the commands package. I just view it as a baby step before _actually_ breaking it out into a package. 

A new package means introducing a folder, and introducing imports for each file. If we ever decide to change the name of the folder, or not make it a package at all, we have to go back and update each reference. Just leaving it inside of `commands` lets us be a lot more aggressive with our refactorings since it's already imported.

I'm not strongly opinionated one way or the other, but just speaking from experience dealing with imports while the API surface is still young can be a little annoying.